src/crimson/osd/CMakeLists.txt

```diff
@@ -33,7 +33,7 @@ add_executable(crimson-osd
   watch.cc
   )
 target_link_libraries(crimson-osd
-  crimson-common crimson-os crimson fmt::fmt)
+  ceph-common crimson-common crimson-os crimson fmt::fmt)
 set_target_properties(crimson-osd PROPERTIES
```

Comment:
Looks like this addition does not have to do anything with shared libs.
Unless it is the way to add `shared_lib` code.
if not you might want te remove it here, or at least make it a separate commit..

Reply:
``shared_lib`` is part of ``ceph-common``, for which reason I had to link ``ceph-osd`` against it, otherwise it won't be able to use those functions.